### DISCUSSION TITLE
UI Grid Table Column Change Non Fill

### DISCUSSION THREAD

**Discussion by grithin** [rexrainbow/phaser3-rex-notes#381] [Q&A]
From example https://codepen.io/rexrainbow/pen/XyJbWX

On changing the column count from 2 to 1
```js
let table = tableScene.table.getElement('table')
table.setColumnCount(1)
table.updateTable(true);
tableScene.table.refresh()
```
The cells now take up 1/2 the width instead of filling the available width.

To get around this, I manually resized the underlying table
```js
let table = tableScene.table.getElement('table')
let width = table.tableWidth
let height = tableScene.table.proportionLength
table.setColumnCount(1)
table.resize(width, height-1)
```
I find if I use proportionLength exactly, it seems to trigger a recalc to the 1/2 width state.

Is there a better way to do this?

**Comment by rexrainbow** [comment]
Sorry I miss this issue totally.

I had updated GridTable plugin, to fix a bug of reset column-width. [Here is a test case](https://codepen.io/rexrainbow/pen/BaEYMPv), line  115-116.

```js
var table = gridTable.getElement('table');
table.setColumnCount(1).updateTable(true);
```

NPM package will upgrade at end of this month.
